File: frontend/src/app/zynerator/security/controller/service/User.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {UserDto} from 'src/app/zynerator/security/controller/model/User.model';
import {UserCriteria} from 'src/app/zynerator/security/controller/criteria/UserCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';
import {RoleService} from './Role.service';


@Injectable({
  providedIn: 'root'
})
export class UserService extends AbstractService<UserDto, UserCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'user/');
    }

    findByUsername(username: string): Observable<UserDto> {
        return this.http.get<UserDto>( this.API + 'Username/' + username);
    }
    public constrcutDto(): UserDto {
        return new UserDto();
    }

    public constrcutCriteria(): UserCriteria {
        return new UserCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/zynerator/security/service/facade/UserService.java
package ma.zs.zyn.zynerator.security.service.facade;

import ma.zs.zyn.zynerator.security.dao.criteria.core.UserCriteria;
import ma.zs.zyn.zynerator.security.bean.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import ma.zs.zyn.zynerator.service.IService;

public interface UserService extends IService<User, UserCriteria>, UserDetailsService {

    User findByUsername(String username);

    User findByUsernameWithRoles(String username);

    String cryptPassword(String value);

    boolean changePassword(String username, String newPassword);

    int deleteByUsername(String username);

    UserDetails loadUserByUsername(String username);


}
